### PR TITLE
doc: boards: extensions: introduce zephyr:board role and directive

### PR DESCRIPTION
**Pull request by kartben**
https://builds.zephyrproject.io/zephyr/pr/79994/docs/boards/seeed/wio_terminal/doc/index.html
https://builds.zephyrproject.io/zephyr/pr/79994/docs/contribute/documentation/guidelines.html#boards

A new `zephyr:board::` Sphinx directive allows to flag a documentation page as being the documentation for a specific board, allowing to auto-populate some of the contents, ex. by adding a board overview *à la Wikipedia*, and later things like supported HW features, a ready-to-copy-paste example of how to build Hello World, etc.

image

A corresponding `:zephyr:board:` role allows to link to a board doc page.

### CODE REVIEW

File: doc/_extensions/zephyr/domain/__init__.py [hunk 1]
```diff
@@ -213,6 +219,64 @@ def convert_node(self, node):
...
+            new_section += nodes.title(text=node["full_name"])
+
+            # create a sidebar with all the board details
+            sidebar = nodes.sidebar(classes=["board-overview"])
+            new_section += sidebar
+            sidebar += nodes.title(text="Board Overview")
+
+            if node["image"] is not None:
+                figure = nodes.figure()
+                figure += nodes.image(uri=f"/{node['image']}")
```
Comment (fabiobaltieri) [inline doc/_extensions/zephyr/domain/__init__.py:247]: cool stuff, should the small image be linked to the full size image? we don't really do high res but looking at the page it's tempting to click (not even sure this is the right place to be honest, I'm talking about the board overview card in the board page https://builds.zephyrproject.io/zephyr/pr/79994/docs/boards/seeed/wio_terminal/doc/index.html)

Comment (kartben) [reply doc/_extensions/zephyr/domain/__init__.py:247]: Ya it's the right place :) and yes, great suggestion

File: doc/templates/board.tmpl [hunk 2]
```diff
@@ -1,20 +1,15 @@
-.. _boardname_linkname:
+.. zephyr:board:: board_name
 
-[Board Name]
-#############
+.. In order for the board documentation page to show properly in the
```
Comment (sylvioalves) [inline doc/templates/board.tmpl:3]: @kartben overall LGTM, I would just improve this text to:
"To ensure the board documentation page displays correctly, it is highly recommended to include a picture of the board."

**Review by gmarull — DISMISSED** [review]

**Comment by gmarull** [comment]
@kartben unrelated to this PR, but, could we make header less cluttered?

image

**Comment by pdgendt** [comment]
> @kartben unrelated to this PR, but, could we make header less cluttered?
> 
> image

The "Open on github" and "Report an issue with this page" could simply be icons IMO, with the text in a tooltip. Of course accessibility needs to be verified.

**Comment by kartben** [comment]
> @kartben unrelated to this PR, but, could we make header less cluttered?
> 
> image

Agreed -- I too found it particularly striking when making the screen capture :) Will go for text-less version of the GH buttons and will try to find a way to have ellipses in the breadcrumb

File: doc/_extensions/zephyr/domain/__init__.py (hunk 1 above)
Comment (kartben) [reply doc/_extensions/zephyr/domain/__init__.py]: > which I think might actually be a bug in RTD theme, btw, so I'll try to submit a fix upstream as well

https://github.com/readthedocs/sphinx_rtd_theme/pull/1622

**Review by carlescufi — DISMISSED** [review]
Really nice stuff!

**Review by erwango — DISMISSED** [review]

**Comment by kartben** [comment]
Just pushed a fix for a nasty bug that was causing Sphinx logs to be silenced - whoopsie.

**Review by fabiobaltieri — DISMISSED** [review]
Great stuff can't wait to see how this renders in the pdf.

**Review by sylvioalves — DISMISSED** [review]

**Comment by kartben** [comment]
> Great stuff can't wait to see how this renders in the pdf.

We don't include boards in the PDF 😝

**Comment by kartben** [comment]
Had to push one more commit to clean up 3.7 release notes re: Sphinx lint.
Just like code samples, boards are now referenced using a dedicated role and not the generic `:ref:` anymore so this requires updates tree-wide, and I deliberately hadn't made a pass on past release notes regarding Sphinx linting as I didn't want to "touch" them unnecessarily (but since the lint errors were actual bugs causing bad rendering, I probably should have...)

**Review by gmarull — APPROVED** [review]
LGTM. It would be nice to start a mass cleanup of board pages. They contain tons of redundant information that just increases doc build time.

**Review by pdgendt — APPROVED** [review]

**Comment by kartben** [comment]
> LGTM. It would be nice to start a mass cleanup of board pages. They contain tons of redundant information that just increases doc build time.

@gmarull this will be incremental and take a lot of time but... yes, definitely -- e.g. https://github.com/zephyrproject-rtos/zephyr/pull/80048

File: doc/releases/release-notes-3.7.rst [hunk 3]
```diff
@@ -834,15 +834,15 @@ Drivers and Sensors
   * Added support for Ambiq Apollo3 series.
   * Added support for multiple instances of the SPI NOR driver (spi_nor.c).
   * Added preliminary support for non-erase devices with introduction of
-    device capabilities to c:struct:`flash_parameters` and the utility function
```
Comment (MaureenHelm) [inline doc/releases/release-notes-3.7.rst:837]: Kind of weird to be modifying release notes of an already finalized release.

Comment (kartben) [reply doc/releases/release-notes-3.7.rst:837]: See https://github.com/zephyrproject-rtos/zephyr/pull/79994#issuecomment-2423233033 for some context

**Review by fabiobaltieri — APPROVED** [review]

**Review by aescolar — APPROVED** [review]

**Pull request merged**